libnd4j/include/ops/declarable/generic/broadcastable/multiply.cpp

```diff
@@ -16,160 +16,158 @@
-            } else {
-                // broadcast case
+namespace ops {
+    
+CUSTOM_OP_IMPL(multiply, 2, 1, true, 0, 0) {
+    
+    NDArray<T> *x = INPUT_VARIABLE(0);
+    NDArray<T> *y = INPUT_VARIABLE(1);
+    NDArray<T> *z = OUTPUT_VARIABLE(0);
+
+    Nd4jLong* zShapeInfo = nullptr;
+    const bool areShapesBroadcastable = ShapeUtils<T>::evalBroadcastShapeInfo(x->getShapeInfo(), y->getShapeInfo(), true, zShapeInfo, block.getWorkspace());
+    REQUIRE_TRUE(areShapesBroadcastable, 0, "MULTIPLY OP: the shapes of x %s and y %s are not suitable for broadcast !", ShapeUtils<T>::shapeAsString(x).c_str(), ShapeUtils<T>::shapeAsString(y).c_str());
+    RELEASE(zShapeInfo, block.getWorkspace());
+
+    z->assign(*x * *y); 
+    return Status::OK();
+}
+
+DECLARE_SYN(Mul, multiply);
+
+DECLARE_SHAPE_FN(multiply) {
+        
+    Nd4jLong* xShapeInfo = inputShape->at(0);
+    Nd4jLong* yShapeInfo = inputShape->at(1);
+
+    Nd4jLong* zShapeInfo = nullptr;
+    const bool areShapesBroadcastable = ShapeUtils<T>::evalBroadcastShapeInfo(xShapeInfo, yShapeInfo, true, zShapeInfo, block.getWorkspace());
+    REQUIRE_TRUE(areShapesBroadcastable, 0, "MULTIPLY OP: the shapes of x %s and y %s are not suitable for broadcast !", ShapeUtils<T>::shapeAsString(xShapeInfo).c_str(), ShapeUtils<T>::shapeAsString(yShapeInfo).c_str());
+
+    return SHAPELIST(zShapeInfo);
+}
+
+
+///////////////////////////////////////////////////////////////////
+CUSTOM_OP_IMPL(multiply_bp, 3, 2, false, 0, 0) {
+
+    NDArray<T>* x    = INPUT_VARIABLE(0);
+    NDArray<T>* y    = INPUT_VARIABLE(1);
+    NDArray<T>* dLdz = INPUT_VARIABLE(2);
+
+    NDArray<T>* dLdx = OUTPUT_VARIABLE(0);
+    NDArray<T>* dLdy = OUTPUT_VARIABLE(1);
+
+    Nd4jLong* dLdzShapeInfo = nullptr;
+    const bool areShapesBroadcastable = ShapeUtils<T>::evalBroadcastShapeInfo(x->getShapeInfo(), y->getShapeInfo(), true, dLdzShapeInfo, block.getWorkspace());
+    REQUIRE_TRUE(areShapesBroadcastable, 0, "MULTIPLY_BP OP: the shapes of x %s and y %s are not suitable for broadcast !", ShapeUtils<T>::shapeAsString(x).c_str(), ShapeUtils<T>::shapeAsString(y).c_str());
+    REQUIRE_TRUE(ShapeUtils<T>::shapeAsString(dLdz) == ShapeUtils<T>::shapeAsString(dLdzShapeInfo), 0, "MULTIPLY_BP OP: wrong shape of next epsilon array (dLdOut), expected is %s, but got %s instead !", ShapeUtils<T>::shapeAsString(dLdzShapeInfo).c_str(), ShapeUtils<T>::shapeAsString(dLdz).c_str());
+    RELEASE(dLdzShapeInfo, block.getWorkspace());
+
+    helpers::multiplyBP<T>(*x, *y, *dLdz, *dLdx, *dLdy);
+
+    return Status::OK();
+}
+
+DECLARE_SHAPE_FN(multiply_bp) {
+    
+    auto xShapeInfo    = inputShape->at(0);
+    auto yShapeInfo    = inputShape->at(1);
+
+    Nd4jLong *dLdxShapeInfo = nullptr;
+    Nd4jLong *dLdyShapeInfo = nullptr;
+
+    COPY_SHAPE(xShapeInfo, dLdxShapeInfo);
+    COPY_SHAPE(yShapeInfo, dLdyShapeInfo);
+
+    return SHAPELIST(dLdxShapeInfo, dLdyShapeInfo);     
+}
+
+        // CUSTOM_OP_IMPL(multiply_bp, 3, 2, false, 0, 0) {
```

Comment:
Please do not do this. Use block comments instead.

Reply:
done